### ISSUE TITLE
Unable to retrieve significant interactions

### ISSUE THREAD

**Issue by ximbao** [ay-lab/FitHiChIP#22]
Hello! I'm trying to run FitHiChIP and I'm getting errorr when calling significant interactions (related to missing or infinite values in inputs). I tried to figured it out myself but I wasn't able to trace back this error.

Here's an example of the output after running it: https://pastebin.com/7nviTJve
Here's the summary of the parameters: https://pastebin.com/kFGWbTAy

I can provide more information if needed! Thank you so much!

**Comment by ay-lab** [comment]
Hi ximbao,
Thanks for your feedback. First of all, please ensure that you are using the latest FitHiChIP release (version 7.0, major update on April 2019). Your configuration file seems to be from older version. New configuration files are free from specifying reference fasta and mappability information. If you are using an older version of FitHiChIP, please re-run your code and configuration file using the latest version.
Considering your issue, can you please check the file 'Bin_Info.log' within the same directory containing FitHiChIP significant interactions, and find any instance of zero or infinite values? You may also mail me the file so that I can take a look into it.

**Comment by ximbao** [comment]
Thank you for your reply!

I updated the pipeline and I'm not specifying the reference fasta anymore, but it's not working.
The file "Bin_Info.log" looks like this:

AvgIntDist	NumLociNonZeroContact	NumLoci_AllPossible	NumContact	AvgContact	PriorProb
NA	26974	0	1453914	Inf	Inf
NA	21857	0	1116882	Inf	Inf
NA	18548	0	908429	Inf	Inf
NA	15825	0	762140	Inf	Inf
NA	13844	0	629850	Inf	Inf
NA	12359	0	559743	Inf	Inf
NA	11314	0	515629	Inf	Inf
NA	10316	0	463476	Inf	Inf
NA	9477	0	428890	Inf	Inf

I've been trying to figure out why that's happening in the past few days...
I can provide the files I'm using to run the pipeline if that'd help troubleshoot this problem.

Thanks again,

**Comment by ay-lab** [comment]
You can send me the files. But first, please check your input interaction files (valid pairs, interval and matrix files) like 1) FitHiChIP.interactions.initial.bed, 2) FitHiChIP.cis.interactions.DistThr.bed, 3) Interactions.bed, 4) Interactions.sortedGenDist.bed (please check documentation for the detailed description and location of these files), and check whether the first 7 fields are of the format (tab delimited):
chr1   start1    end1   chr2   start2  end2   cc
without any errors. Also, whether the chromosomes are specified as chr1, chr2, instead of numbers like 1, 2, ...
I think there is some formatting error within one of these files, resulting the interaction distance (or even the possible locus pairs) to be NA or 0.
Also, did you run with the new code and without any reference fasta / mappability / GC content files, and using OverWrite = 1 option? Please run if you have not done so. This is to check whether the reference fasta / gc content / mappability information is causing any changes in the format.

**Comment by GeorgeBGM** [comment]
Hello!I also got similarly error message using my data produced by _HiCPro_ software.I ran my data using BiasCorrection model with the follow set, But  I got reasonable results when I using TestData of _FitHiChIP_. So can you give me several advises.
#################
ValidPairs=./hic_results/data/CTCF_FKDL190766179-1a/CTCF_FKDL190766179-1a.allValidPairs
ChrSizeFile=pseudohap2_1_2.filt.hap1.chrom.size
PeakFile=./PeakInferHiChIP/MACS2_ExtSize/out_macs2_peaks.narrowPeak
#################
I am looking forward to your reply.Thanks.

**Comment by ay-lab** [comment]
Hi @ximbao  and @George-du 

Thanks for your queries. Please check the file "Bin_Info.log" within the FitHiChIP output directory. If the first column contains NA (as @ximbao already mentioned), please check if your valid pairs file contains chromosome information as 1, 2, .. instead of chr1, chr2, ... In such a case, please replace 1, 2, with chr1, chr2, .. and re-run FitHiChIP.

If that does not solve your problem, please mail me your configuration file, Bin_Info.log file, and the  console output summary.

**Comment by ay-lab** [comment]
Hi @ximbao and @George-du

Please inform if the problems are solved. Also, I have updated FitHiChIP code today, so please clone with the GitHub repository. For any problems, please inform me.

**Comment by shelbyar** [comment]
Hello! Thank you for the super useful tool. I'm currently having a similar issue to this thread, where I get no significant interactions and NA/infinite values in the bin_info.log file (despite test data running well). My interactions.bed file seems to be formatted correctly but only contains a subset of chromosome 3. I'm using the newest version of fithichip and I'm directly using the allvalidpairs file from hic-pro for input. I've tried 5, 10, and 20kb bins with peak2all and loose and stringent backgrounds. I've attached the output and error I'm seeing (edited with ** to replace directories). Do you have any thoughts on what might be going wrong? 
[fithichip_1.txt](https://github.com/ay-lab/FitHiChIP/files/8828702/fithichip_1.txt)
[fithichip_1error.txt](https://github.com/ay-lab/FitHiChIP/files/8828703/fithichip_1error.txt)

**Comment by ay-lab** [comment]
Hi @shelbyar 
Thanks for your query. The log file shows "Total Number of interactions: 86" which is too few. The spline fitting and regression estimates cannot work with such a few entries. Could you please try using all the interactions in chr3 and using P2P=0 background? Increasing the number of interactions should solve the problem.
On a different note, instead of loop calling chromosome-wise, I recommend using the complete set of contacts (from all chromosomes) at once.

**Comment by shelbyar** [comment]
Thank you for the quick response! I think I was unclear- I did attempt to run this on all chromosomes, but it seems like there has been an error and I noticed that the intermediate interactions.bed produced by fithichip only contains part of chr3. My input files contain all chromosomes, though.  edit to add: In another pipeline where I'm using the same allvalidpairs hic-pro output, I ran into an issue due to the inclusion of small/alt contig chromosomes in my allvalidpairs. Could that be part of the problem?
